### PR TITLE
[UI] [Integrate] Reward Detail

### PR DESCRIPTION
**Pull request by ducbm051291**
## What happened 👀

- Add Reward Detail API
- Create function to get reward detail in RewardRepository
- Add function to get reward detail in NimbleLoyalty
- Create and implement RewardDetail (View and ViewModel)
- Create NavigationBackButtonModifier to custom back button

## Insight 📝

- n/a

## Proof Of Work 📹

https://user-images.githubusercontent.com/24598204/227848606-05fd5f32-8090-4507-8d87-db0a94ce84e1.mp4

### CODE REVIEW

**Comment by suho** [comment]
@ducbm051291 There are duplicated code with another pull request, please check the base branch of this PR

**Comment by ducbm051291** [comment]
> @ducbm051291 There are duplicated code with another pull request, please check the base branch of this PR

Actually, the merge from Main to Develop must be done before this. I will convert it to draft and update the README

File: Sources/NimbleLoyalty.swift [hunk 1]
```diff
@@ -105,6 +105,21 @@ extension NimbleLoyalty {
         }
     }
 
+    public func getRewardDetail(code: String, _ completion: @escaping (Result<APIReward, NimbleLoyaltyError>) -> Void) {
```
Comment (suho) [inline Sources/NimbleLoyalty.swift:108]: @ducbm051291 You could update README about get reward detail in this PR

Comment (ducbm051291) [reply Sources/NimbleLoyalty.swift:108]: I've already updated in this PR

**Review by suho — APPROVED** [review]

**Pull request merged**